### DISCUSSION TITLE
Exception thrown when using overwriteRequestSecurity

### DISCUSSION THREAD

**Discussion by PaulCluver** [apideck-libraries/portman#507] [Q&A] [answered]
Hi there, 

I am having a problem using overwriteRequestSecurity which I hope someone can assist with. The details of this issue are as follows: 

**Configuration**
![image](https://github.com/apideck-libraries/portman/assets/5680263/f21affc1-16f5-45d2-ac35-4f6b75850320)

**Exception**
![image](https://github.com/apideck-libraries/portman/assets/5680263/2768ddd5-0667-4ba1-9fc2-bc6d9ad8cb74)

In debugging this, it appears that var authConfig is not getting set. 
![image](https://github.com/apideck-libraries/portman/assets/5680263/5371f605-dd03-4569-95fe-8dc0f84490d3)

Stepping into method getAuth() and as per comments, the method looks for auth on request.  If this is not found or is empty, the method searches for the auth on the parent. This is where I believe that I am coming unstuck: 

![image](https://github.com/apideck-libraries/portman/assets/5680263/d1937767-9feb-4133-b4ff-d0710cb25a0b)

Authorization is being set at the parent level, so I am not too sure why getAuth() is not working. 

![image](https://github.com/apideck-libraries/portman/assets/5680263/93bb5bd1-d3da-42a8-8b2b-affa407fc5ee)

If anyone has had any ideas on how to get around this, it would be most appreciated. 

Kind regards,
Paul.

Kind regards,
Paul

**Comment by thim81** [comment]
@PaulCluver Is there an option that you can share your OpenAPI document? So we can use that to reproduce the issue?

**Reply by PaulCluver** [reply]
Hi, 

Let me confer with one of my colleagues first. I don't really see a problem in sharing this but I do need to check first. I will revert back as soon as possible. What I can comfortably share at this point is as follows: 

![image](https://github.com/apideck-libraries/portman/assets/5680263/957360c4-3534-4c92-8790-6d71655cac0c)

This is all that I can find with respect to authentication in the api spec. 

In the generated postman collection, the auth is as follows: 

![image](https://github.com/apideck-libraries/portman/assets/5680263/dff39d33-f705-4a75-a88d-f43271e6f439)

Kind regards,
Paul.

**Reply by thim81** [reply, marked as answer]
hi @PaulCluver 

This issue was fixed in version [v1.23.1](https://github.com/apideck-libraries/portman/blob/main/CHANGELOG.md#v1231---2023-10-28)
